src/util-base64.c

```diff
@@ -83,14 +83,21 @@ bool IsBase64Alphabet(uint8_t encoded_byte)
  *
  * \param ascii the 3-byte ascii output block
  * \param b64 the 4-byte base64 input block
+ * \param nb number of decoded bytes
  *
  * \return none
  */
-static inline void DecodeBase64Block(uint8_t ascii[ASCII_BLOCK], uint8_t b64[B64_BLOCK])
+static inline void DecodeBase64Block(uint8_t ascii[ASCII_BLOCK], uint8_t b64[B64_BLOCK], size_t nb)
```

Comment:
Other solution is to ensure `DecodeBase64` is called with a big enough buffer...